CHANGELOG.md

```diff
@@ -1,3 +1,431 @@
+    ([#735](https://github.com/kinvolk/lokomotive/pull/735)).
+
+- Add `codespell` target to Makefile and `all` target ([#661](https://github.com/kinvolk/lokomotive/pull/661)).
+
+### Upgrading from v0.2.1
+
+#### Configuration syntax changes
+
+There have been some minor changes to the configurations of following components:
+* contour
+* rometheus-operator.
+
+Please make sure new the configuration structure is in place before the upgrade.
+
+##### Contour component
+
+Optional `ingress_hosts` variable is now removed.
+
+old:
+
+```hcl
+component "contour" {
+  .
+  .
+  ingress_hosts = ["*.example.lokomotive-k8s.net"]
+}
+```
+
+new:
+
+```hcl
+component "contour" {
+  .
+  .
+}
+
+```
+
+##### Prometheus-operator component
+
+* Variable `external_url` is now removed.
+* Prometheus specific variables are now under a `prometheus` block.
+* A new optional `prometheus.ingress` sub-block is introduced to expose
+Prometheus over ingress.
+
+old:
+
+```hcl
+component "prometheus-operator" {
+  .
+  .
+  prometheus_metrics_retention = "14d"
+  prometheus_external_url      = "https://api.example.com/prometheus"
+  prometheus_storage_size      = "50GiB"
+  prometheus_node_selector = {
+    "kubernetes.io/hostname" = "worker3"
+  }
+  .
+  .
+}
+```
+
+new:
+
+```hcl
+component "prometheus-operator" {
+  .
+  .
+  prometheus {
+    metrics_retention = "14d"
+    storage_size      = "50GiB"
+    node_selector = {
+      "kubernetes.io/hostname" = "worker3"
+    }
+
+    ingress {
+      host                       = "prometheus.example.lokomotive-k8s.net"
+      class                      = "contour"
+      certmanager_cluster_issuer = "letsencrypt-production"
+    }
+    .
+    .
+  }
+  .
+  .
+}
+```
+
+Check out the new syntax in the [Prometheus Operator configuration
+reference](https://github.com/kinvolk/lokomotive/blob/v0.2.0/docs/configuration-reference/components/prometheus-operator.md)
+for details.
+
+##### Upgrade steps
+
+Go to your cluster's directory and run the following command.
+
+```
+lokoctl cluster apply
+```
+
+The update process typically takes about 10 minutes.
+After the update, running `lokoctl health` should result in an output similar to the following.
+
+```
+Node                     Ready    Reason          Message
+
+lokomotive-controller-0  True     KubeletReady    kubelet is posting ready status
+lokomotive-1-worker-0    True     KubeletReady    kubelet is posting ready status
+lokomotive-1-worker-1    True     KubeletReady    kubelet is posting ready status
+lokomotive-1-worker-2    True     KubeletReady    kubelet is posting ready status
+Name      Status    Message              Error
+
+etcd-0    True      {"health":"true"}
+```
+
+##### Post upgrade steps
+
+##### Openebs
+
+>It is highly recommended to schedule a downtime for the application using the
+>OpenEBS PV while performing this upgrade. Also, make sure you have taken a
+>backup of the data before starting the below upgrade procedure.
```

Comment:
Why the quote? For style/format or comes from openebs docs? If it's the latter, I'd reference that

Reply:
Would changing to a **NOTE** from a quote help ?

Openebs docs have upgrade procedures for upgrading the control plane components, jiva storage engine and cstore storage engine. I think pointing to the openebs docs would cause more confusion.

Reply:
I didn't meant pasting the docs, just saying that if this is a quote from openebs docs, add a link to where they come from. Not paste the content, just a link

Reply:
Added the link to the openebs documentation.